### ISSUE TITLE
Error setting up entry Smart Irrigation for smart_irrigation

### ISSUE THREAD

**Issue by andreid1303** [jeroenterheerdt/HAsmartirrigation#214]
### What happened?

After update to  v2023.8.0-beta29 integration does not start

### How to reproduce

Update from v2023.8.0-beta27 to v2023.8.0-beta29, restart HA

### Relevant log output

```shell
Error setting up entry Smart Irrigation for smart_irrigation
Traceback (most recent call last):
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/config_entries.py", line 388, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/__init__.py", line 62, in async_setup_entry
    store = await async_get_registry(hass)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/store.py", line 524, in async_get_registry
    return cast(SmartIrrigationStorage, await task)
                                        ^^^^^^^^^^
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/store.py", line 519, in _load_reg
    await registry.async_load()
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/store.py", line 191, in async_load
    maximum_duration = zone[ZONE_MAXIMUM_DURATION]
                       ~~~~^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'maximum_duration'
```

### Which version are you running?

v2023.8.0-beta29

### Diagnostics file

- [X] I have attached a diagnostics file

### Additional information

Diagnostics not available

**Comment by omenek159** [comment]
@jeroenterheerdt forgot to put default values on start. Quick solution is to edit .storage/smart_irrigation.storage file and for each zone add "maximum_duration": 0 . Don't forget to add comma at the end of  lead_time

![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/10829566/12898fda-8920-4cf3-b2f4-bd4e8d2579f0)

**Comment by andreid1303** [comment]
Ok, thanks, started now. But Calculate All call give me follow error:
Error handling request
Traceback (most recent call last):
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/security_filter.py", line 85, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/view.py", line 148, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/var/homeassistant/homeassistant/lib/python3.11/site-packages/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/websockets.py", line 176, in post
    await coordinator.async_update_zone_config(zone, data)
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/__init__.py", line 732, in async_update_zone_config
    await self._async_calculate_all()
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/__init__.py", line 411, in _async_calculate_all
    data = self.calculate_module(zone, mapping_weatherdata, precip_from_sensor=precip_from_sensor, sol_rad_from_sensor = sol_rad_from_sensor,et_data=et_from_sensor)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/homeassistant/.homeassistant/custom_components/smart_irrigation/__init__.py", line 474, in calculate_module
    data[const.ZONE_BUCKET] = bucket+delta
                              ~~~~~~^~~~~~
TypeError: unsupported operand type(s) for +: 'NoneType' and 'float'

**Comment by andreid1303** [comment]
And Duration is not calculated

[config_entry-smart_irrigation-6e18a9ad38f8ac00ef4285672ad17839.json-3.txt](https://github.com/jeroenterheerdt/HAsmartirrigation/files/12495024/config_entry-smart_irrigation-6e18a9ad38f8ac00ef4285672ad17839.json-3.txt)

**Comment by omenek159** [comment]
For me calculation don't throw any error, but I noticed that putting 0 in maximum_duration causes to Duration be always 0. Change it to null .

**Comment by andreid1303** [comment]
Understood now. Changed Maximum Duration to required value of 3600 s, reset and calculate buckets, no errors in logs. You may close this issue.

**Comment by jeroenterheerdt** [comment]
Yeah, maximum duration should have been null in the storage by default but it did not work unfortunately. I need to look at this once more to try and make it appear by default without giving it a value so the fix to edit the storage is not necessary

**Comment by andreid1303** [comment]
Why not 60min or 3600 s as default, for example?

**Comment by jeroenterheerdt** [comment]
Well yes, that is something I can do!

**Comment by jeroenterheerdt** [comment]
should be fixed in v2023.9.0-beta1
